src/schema/v2/filterArtworksConnection.ts

```diff
@@ -575,39 +573,57 @@ const filterArtworksConnectionTypeFactory = (
         )
       }
 
-      const totalPages = computeTotalPages(
-        aggregations.total.value,
-        gravityOptions.size
+      const hasRequestedCollectorSignals = isFieldRequested(
+        "edges.node.collectorSignals",
+        info
       )
 
-      const connection = connectionFromArraySlice(
-        hits,
-        { first, last, after, before },
-        {
-          arrayLength: Math.min(
-            aggregations.total.value,
-            totalPages * gravityOptions.size
-          ),
-          sliceStart: gravityOptions.offset,
-        }
-      )
+      const collectorSignalsRequests = hasRequestedCollectorSignals
+        ? hits.map((artwork) => {
+            return fetchCollectorSignals(artwork, ctx).then(
+              (collectorSignals) => {
+                artwork.collectorSignals = collectorSignals
+                return artwork
+              }
+            )
+          })
+        : hits
+
+      return Promise.all(collectorSignalsRequests).then((artworks) => {
```

Comment:
#non-blocking (but nice 😉 ): can we use async await here vs chained promises? 

```
try {
  const artworks = await Promise.all(collectorSignalRequests)
  ...
} catch (error) {
  // handle / return error 
}

Reply:
I actually want to caution against zealous `catch`. In MP, without any `catch` - you get pretty desirable behavior. There is Sentry reporting, the error shows up in `errors` in the response and the corresponding field is `null`, and the error in the response includes relevant information to help debug.

There are a lot of examples of `catch` that simply log 'Error fetching .... ' -> not actually providing any information, and the resulting field is usually still `null`, thus `catch` does nothing there. Or `catch` and throwing your own error there - more or less the same thing.

Basically you _shouldn't_ need a `catch` in most cases.

Reply:
Yah, i mean it (should) be understood that you have to return the error! but i know what you mean. Still should avoid chained vanilla promises. We can let it error out without try / catch.

Reply:
👍 Updated the whole resolver to use async/await _except_ this little `hits.map` because it gave me a [probably incorrect] linter error `Possible race condition: `artwork.collectorSignals` might be reassigned based on an outdated value of `artwork.collectorSignals`.eslintrequire-atomic-updates`
![image](https://github.com/user-attachments/assets/a5e2cbbd-f9f6-479c-a145-b4596644a294)

So that part being quite small I just left as a chained promise.